### PR TITLE
Refactor bigquery tests into a separate module

### PR DESCRIPTION
**Pull request by abhishoya-gs**
#### What type of PR is this?
 
Test Enhancement

#### What does this PR do / why is it needed ?

- Introduce a new module for BiqQuery tests
- Add support for SQL query tests against BigQuery and enable them in Github workflows

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

No

### CODE REVIEW

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/tests/api/dynamicTestConnections/BigQueryDynamicConnection.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.finos.legend.engine.plan.execution.stores.relational.connection.tests.api.DynamicTestConnection;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.DatabaseType;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.RelationalDatabaseConnection;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.authentication.GCPWorkloadIdentityFederationAuthenticationStrategy;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.specification.BigQueryDatasourceSpecification;
+import org.finos.legend.engine.shared.core.vault.EnvironmentVaultImplementation;
+import org.finos.legend.engine.shared.core.vault.Vault;
+import org.finos.legend.engine.shared.core.vault.VaultImplementation;
+
+public class BigQueryDynamicConnection implements DynamicTestConnection
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-bigquery-execution-tests/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/tests/api/dynamicTestConnections/BigQueryDynamicConnection.java:26]: This is not a dynamic test connection, but a static one. Delete this file and create a json file, containing the connection details

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/main/resources/META-INF/services/org.finos.legend.engine.plan.execution.stores.relational.connection.tests.api.DynamicTestConnection [hunk 2]
```diff
@@ -0,0 +1 @@
+org.finos.legend.engine.plan.execution.stores.relational.connection.tests.api.dynamicTestConnections.BigQueryDynamicConnection
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-bigquery-execution-tests/src/main/resources/META-INF/services/org.finos.legend.engine.plan.execution.stores.relational.connection.tests.api.DynamicTestConnection:1]: remove this file

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/resources/org/finos/legend/engine/server/test/userTestConfig_withBigQueryTestConnection.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      },
+      "gcpWorkloadConfig": {
+        "projectNumber": "412074507462",
+        "poolId": "integration-wif-pool1",
+        "providerId": "integration-wif-pool1-provider"
+      }
+    }
+  },
+  "testConnectionsToEnable" : ["bigQuery"],
+  "dynamicTestConnectionCreators": {
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/resources/org/finos/legend/engine/server/test/userTestConfig_withBigQueryTestConnection.json:56]: it should have static test connections field instead of dynamic

File: pom.xml [hunk 4]
```diff
@@ -184,6 +184,7 @@
         <module>legend-engine-testable</module>
         <module>legend-engine-executionPlan-execution-authorizer</module>
         <module>legend-engine-xt-relationalStore-executionPlan-authorizer</module>
+        <module>legend-engine-xt-relationalStore-bigquery-execution-tests</module>
```
Comment (vikask-gs) [inline pom.xml:187]: add this line in the place where all the other big query modules are listed

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/protocols/pure/vX_X_X/invocations/execution_relational_testConnection.pure [hunk 5]
```diff
@@ -63,7 +63,7 @@ function meta::protocols::pure::vX_X_X::transformation::toPureGraph::connection:
                                                             pair('snowflakePublic', 'SnowflakePublicAuthenticationStrategy'),
                                                             pair('gcpApplicationDefaultCredentials', 'GCPApplicationDefaultCredentialsAuthenticationStrategy'),
                                                             pair('apiToken', 'ApiTokenAuthenticationStrategy'),
-                                                            pair('gcpWorkloadIdentityFederationWithAWS','GCPWorkloadIdentityFederationWithAWSAuthenticationStrategy'),
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/protocols/pure/vX_X_X/invocations/execution_relational_testConnection.pure:66]: why this change?

File: legend-engine-xt-relationalStore-bigquery-pure/src/main/resources/core_relational_bigquery/relational/tests/dbSpecificTests/bigQueryTests.pure [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  meta::relational::tests::dbSpecificTests::sqlQueryTests::dynaFunctions
+          ->collectParameterizedTests('BigQuery',
+                ^DbTestConfig(dbType=DatabaseType.BigQuery, connection=$connection), [],
+                meta::relational::tests::dbSpecificTests::bigQuery::testsFilter_FunctionDefinition_1__Boolean_1_
+                );
+}
+
+function meta::relational::tests::dbSpecificTests::bigQuery::testsFilter(f:FunctionDefinition<Any>[1]):Boolean[1]
+{
+  // currently hardcoding unsupported funcs, In future fetch this from db extensions and skip
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-bigquery-pure/src/main/resources/core_relational_bigquery/relational/tests/dbSpecificTests/bigQueryTests.pure:34]: start by keeping it as simply true, and then add skip based on integration test run

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java [hunk 7]
```diff
@@ -43,7 +43,7 @@
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
-public class ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery extends DbSpecificTests
+public class ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery extends RelationalConnectionTest
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java:46]: Remove the connection details from code and instead load it from a json file containing the connection. Also delete all the big query tests outside this module. If you search for bigquery across the legend-engine project, only modules containing bigquery in their name, should have bigquery code.

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/tests/api/dynamicTestConnections/BigQueryDynamicConnection.java (hunk 1 above)
Comment (abhishoya-gs) [reply legend-engine-xt-relationalStore-bigquery-execution-tests/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/tests/api/dynamicTestConnections/BigQueryDynamicConnection.java]: done

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/main/resources/META-INF/services/org.finos.legend.engine.plan.execution.stores.relational.connection.tests.api.DynamicTestConnection (hunk 2 above)
Comment (abhishoya-gs) [reply legend-engine-xt-relationalStore-bigquery-execution-tests/src/main/resources/META-INF/services/org.finos.legend.engine.plan.execution.stores.relational.connection.tests.api.DynamicTestConnection]: done

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/resources/org/finos/legend/engine/server/test/userTestConfig_withBigQueryTestConnection.json (hunk 3 above)
Comment (abhishoya-gs) [reply legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/resources/org/finos/legend/engine/server/test/userTestConfig_withBigQueryTestConnection.json]: changed

File: pom.xml (hunk 4 above)
Comment (abhishoya-gs) [reply pom.xml]: done

File: legend-engine-xt-relationalStore-bigquery-pure/src/main/resources/core_relational_bigquery/relational/tests/dbSpecificTests/bigQueryTests.pure (hunk 6 above)
Comment (abhishoya-gs) [reply legend-engine-xt-relationalStore-bigquery-pure/src/main/resources/core_relational_bigquery/relational/tests/dbSpecificTests/bigQueryTests.pure:34]: yes, I marked these tests as failed and unsupported after running them locally

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/protocols/pure/vX_X_X/invocations/execution_relational_testConnection.pure (hunk 5 above)
Comment (vikask-gs) [reply legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/protocols/pure/vX_X_X/invocations/execution_relational_testConnection.pure:66]: ok

File: legend-engine-xt-relationalStore-bigquery-pure/src/main/resources/core_relational_bigquery/relational/tests/dbSpecificTests/bigQueryTests.pure (hunk 6 above)
Comment (vikask-gs) [reply legend-engine-xt-relationalStore-bigquery-pure/src/main/resources/core_relational_bigquery/relational/tests/dbSpecificTests/bigQueryTests.pure:34]: ok

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java (hunk 7 above)
Comment (abhishoya-gs) [reply legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java:47]: Done.

Note: There are tests in authentication module for Auth strategies supported with BigQuery. It doesn't make sense to bring them into BigQuery module as they don't interact with the database itself.

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java [hunk 8]
```diff
@@ -146,41 +128,45 @@ public void testBigQueryGCPWIFConnection_subject() throws Exception
...
 
-    private RelationalDatabaseConnection bigQueryWithGCPWIFSpec()
-    {
-        BigQueryDatasourceSpecification bigQueryDatasourceSpecification = new BigQueryDatasourceSpecification();
-        bigQueryDatasourceSpecification.projectId = "legend-integration-testing";
-        bigQueryDatasourceSpecification.defaultDataset = "integration_dataset1";
-        GCPWorkloadIdentityFederationAuthenticationStrategy authSpec = new GCPWorkloadIdentityFederationAuthenticationStrategy();
-        authSpec.serviceAccountEmail = "integration-bq-sa1@legend-integration-testing.iam.gserviceaccount.com";
-        return new RelationalDatabaseConnection(bigQueryDatasourceSpecification, authSpec, DatabaseType.BigQuery);
+    private RelationalDatabaseConnection bigQueryWithGCPWIFSpec() throws JsonProcessingException {
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java:162]: If you keep different element values for each connection instead of empty string, we can use getRelationalConnectionByElement to select the appropriate one for test

Comment (abhishoya-gs) [reply legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java:158]: changed

File: legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java [hunk 9]
```diff
@@ -86,32 +74,27 @@ public static void verifyTestSetup()
...
     @Before
-    public void setup()
-    {
-        LegendDefaultDatabaseAuthenticationFlowProvider flowProvider = new LegendDefaultDatabaseAuthenticationFlowProvider();
-        LegendDefaultDatabaseAuthenticationFlowProviderConfiguration flowProviderConfiguration = LegendDefaultDatabaseAuthenticationFlowProviderConfiguration.Builder.newInstance()
-                .withAwsConfig(awsConfig)
-                .withGcpWorkloadConfig(gcpWorkloadConfig)
-                .build();
+    public void setup() throws IOException {
+        RelationalConnectionTest rtb = new RelationalConnectionTest();
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java:84]: no need to created rtb objects, as this class extends RelationalConnectionTest, so its protected methods are directly available, like how testConnection is available

**Comment by vikask-gs** [comment]
Good to merge!

**Review by gs-ssh16 — APPROVED** [review]
